src/Setup/WixBA/RootView.xaml

```diff
@@ -1,125 +1,487 @@
-            <Image Grid.Row="2" Grid.Column="2" Source="resources\exit.png" DataContext="{Binding InstallationViewModel}" Visibility="{Binding ExitEnabled, Converter={StaticResource BooleanToVisibilityConverter}}" />
-
-            <!-- Cancel -->
-            <Button Grid.Row="2" Grid.Column="2" Background="#FFCB0000" DataContext="{Binding}" Visibility="{Binding CancelEnabled, Converter={StaticResource BooleanToVisibilityConverter}}" Command="{Binding CancelCommand}">Cancel</Button>
-            <TextBlock Grid.Row="2" Grid.Column="2" HorizontalAlignment="Center" VerticalAlignment="Center" FontSize="60" Foreground="White" IsHitTestVisible="False" DataContext="{Binding}"  Visibility="{Binding CancelEnabled, Converter={StaticResource BooleanToVisibilityConverter}}">X</TextBlock>
-
-        </Grid>
-
-    </Grid>
+    <i:Interaction.Behaviors>
+        <ux:WinNoSnapBehavior />
+    </i:Interaction.Behaviors>
+
+    <Border BorderBrush="{DynamicResource {x:Static SystemColors.ActiveBorderBrushKey}}" BorderThickness="0">
+        <!--  Catch MouseLeftButtonDown for Window dragging.  -->
+        <Border HorizontalAlignment="Stretch"
+                VerticalAlignment="Stretch"
+                Background="{DynamicResource {x:Static SystemColors.WindowBrushKey}}"
+                MouseLeftButtonDown="Background_MouseLeftButtonDown">
+
+            <!--  If click on no-input control (i.e. TextBlock, ProgressBar) catch click here to avoid InvalidOperationException.  -->
+            <DockPanel x:Name="AppArtBoardArea"
+                       VerticalAlignment="Stretch"
+                       MouseDown="RootView_IgnoreClick"
+                       MouseLeftButtonDown="Background_MouseLeftButtonDown"
+                       MouseUp="RootView_IgnoreClick">
+
+                <!--  Update checking marquee, Initial Actions, and Final Action Buttons, at the bottom of the layout.  -->
+                <DockPanel x:Name="ActionArea"
+                           Margin="0,0,10,10"
+                           HorizontalAlignment="Stretch"
+                           DockPanel.Dock="Bottom">
+                    <Grid x:Name="ActionGrid" HorizontalAlignment="Stretch">
+                        <Grid.RowDefinitions>
+                            <RowDefinition Height="auto" />
+                        </Grid.RowDefinitions>
+                        <Grid.ColumnDefinitions>
+                            <ColumnDefinition Width="5" />
+                            <ColumnDefinition Width="auto" />
+                            <ColumnDefinition Width="*" MinWidth="30" />
+                            <ColumnDefinition Width="auto" />
+                            <ColumnDefinition Width="5" />
+                        </Grid.ColumnDefinitions>
+
+                        <StackPanel x:Name="LeftMarqueeDcIsland"
```

Comment:
OK, just wondering.  I think the WPF purists would want each "DCIsland" to be a UserControl, which would result in each view with it's own view model.  I don't really think this UI is large enough for UserControls to be worth it.